formats/dsd/format.go

```diff
@@ -0,0 +1,72 @@
+package dsd
+
+import "errors"
+
+var (
+	ErrIncompatibleFormat = errors.New("dsd: format is incompatible with operation")
+	ErrIsRaw              = errors.New("dsd: given data is in raw format")
+	ErrNoMoreSpace        = errors.New("dsd: no more space left after reading dsd type")
+	ErrUnknownFormat      = errors.New("dsd: format is unknown")
+)
+
+// Format types.
+const (
+	AUTO = 0
+
+	// Serialization types.
+	RAW     = 1
+	CBOR    = 67 // C
+	GenCode = 71 // G
+	JSON    = 74 // J
+	MsgPack = 77 // M
+
+	// Compression types.
+	GZIP = 90 // Z
+
+	// Special types.
+	LIST = 76 // L
+
+	// Deprecated: NONE is deprecated, please use RAW instead.
+	NONE = 1
+)
+
+var (
+	DefaultSerializationFormat uint8 = JSON
+	DefaultCompressionFormat   uint8 = GZIP
+)
+
+// ValidateSerializationFormat validates if the format is for serialization,
+// and returns the validated format as well as the result of the validation.
+// If called on the AUTO format, it returns the default serialization format.
+func ValidateSerializationFormat(format uint8) (validatedFormat uint8, ok bool) {
+	switch format {
+	case AUTO:
+		return DefaultSerializationFormat, true
+	case RAW:
+		return format, true
```

Comment:
`NONE` is still defined so this should consider that:
```suggestion
        case RAW, NONE:
            return RAW, true
```

Reply:
This is not necessary, are RAW and NONE have the same value. I just want to outphase "NONE", as "RAW" is much more descriptive, imho.